### PR TITLE
Add single quotes around pip install command

### PR DESCRIPTION
**Pull request by robnewman**
# Description
Fix all references to `pip install 'quilt3[catalog]'` and `pip install 'quilt[pyarrow]'`

# TODO

- [n/a] Unit tests
- [n/a] Automated tests (e.g. Preflight)
- [X] Documentation
    - [n/a] [Python: Run `build.py`](../gendocs/build.py) for new docstrings
    - [n/a] JavaScript: basic explanation and screenshot of new features
- [n/a] [Changelog](CHANGELOG.md) entry (skip if change is not significant to end users, e.g. docs only)

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ Entries inside each section should be ordered by type:
 * [Added] Declared compatibility with `jsonschema==4.*`.
 * [Added] `--host` and `--port` parameters for `quilt3 catalog`.
 * [Added] `--no-browser` parameter for `quilt3 catalog`.
-* [Changed] `quilt3 catalog` now requires `quilt3` to be installed with `catalog` extra dependency (`pip install quilt3[catalog]`).
+* [Changed] `quilt3 catalog` now requires `quilt3` to be installed with `catalog` extra dependency (`pip install 'quilt3[catalog]'`).
```
Comment (akarve) [inline docs/CHANGELOG.md:112]: No changelog for docs changes, right?

Comment (akarve) [reply docs/CHANGELOG.md:112]: Description is also incorrect. We just need to single quote any instances of the extras installs.

**Review by akarve — DISMISSED** [review]
Need more quotes e.g. here https://docs.quiltdata.com/installation#python-client

**Comment by sir-sigurd** [comment]
Also https://github.com/quiltdata/quilt/blob/0f50ca57edc95e1bd795c28c90de44b3d0146cf9/docs/CONTRIBUTING.md?plain=1#L37.

**Review by sir-sigurd — CHANGES_REQUESTED** [review]
https://github.com/quiltdata/quilt/blob/0f50ca57edc95e1bd795c28c90de44b3d0146cf9/docs/CONTRIBUTING.md?plain=1#L37

**Review by sir-sigurd — APPROVED** [review]
I'd prefer line wrapping changes done separately, but OK.

**Pull request merged**